hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -95,712 +82,510 @@ cardanoLedger =
-  fromJSONKey = FromJSONKeyTextParser txInFromText
-
-instance Crypto crypto => ToJSONKey (Cardano.TxIn crypto) where
-  toJSONKey = toJSONKeyText txInToText
+  liftLovelace :: TxOutValue Era -> Value
+  liftLovelace = \case
+    TxOutAdaOnly _ lovelace -> lovelaceToValue lovelace
+    TxOutValue _ value -> value
+
+  toLedgerDataHash :: TxOutDatum CtxUTxO Era -> StrictMaybe (Ledger.Alonzo.DataHash Ledger.StandardCrypto)
+  toLedgerDataHash = \case
+    TxOutDatumNone -> SNothing
+    TxOutDatumHash _ (ScriptDataHash h) -> SJust h
+
+toMaryTxOut :: Ledger.TxOut LedgerEra -> Ledger.Mary.TxOut (Ledger.Mary.MaryEra Ledger.StandardCrypto)
+toMaryTxOut = \case
+  Ledger.Alonzo.TxOutCompact addr value ->
+    Ledger.Shelley.TxOutCompact addr value
+  Ledger.Alonzo.TxOutCompactDH addr value _datum ->
+    Ledger.Shelley.TxOutCompact addr value
+
+fromMaryTxOut :: Ledger.Mary.TxOut (Ledger.Mary.MaryEra Ledger.StandardCrypto) -> Ledger.TxOut LedgerEra
+fromMaryTxOut = \case
+  Ledger.Shelley.TxOutCompact addr value ->
+    Ledger.Alonzo.TxOutCompact addr value
 
 --
--- Address
+-- KeyWitness
 --
 
-mkVkAddress ::
-  Cardano.VKey 'Cardano.Payment StandardCrypto ->
-  CardanoAddress
-mkVkAddress vk =
-  let paymentCredential = Cardano.KeyHashObj $ Cardano.hashKey vk
-      stakeReference = Cardano.StakeRefNull
-   in Cardano.Addr Cardano.Testnet paymentCredential stakeReference
-
--- Serialise addresses in bech32 including the prefix as standardized:
--- https://github.com/cardano-foundation/CIPs/blob/master/CIP-0005/CIP-0005.md
-encodeAddress ::
-  Cardano.Addr crypto ->
-  Text
-encodeAddress addr =
-  Bech32.encodeLenient prefix . Bech32.dataPartFromBytes $ Cardano.serialiseAddr addr
+toLedgerKeyWitness ::
+  [KeyWitness era] ->
+  Set (Ledger.Shelley.WitVKey 'Ledger.Witness Ledger.StandardCrypto)
+toLedgerKeyWitness vkWits =
+  fromList [w | ShelleyKeyWitness _ w <- vkWits]
+
+toLedgerBootstrapWitness ::
+  [KeyWitness era] ->
+  Set (Ledger.BootstrapWitness Ledger.StandardCrypto)
+toLedgerBootstrapWitness vkWits =
+  fromList [w | ShelleyBootstrapWitness _ w <- vkWits]
+
+fromLedgerTxWitness :: Ledger.Alonzo.TxWitness LedgerEra -> [KeyWitness Era]
+fromLedgerTxWitness wits =
+  Set.foldr ((:) . ShelleyKeyWitness era) [] (Ledger.Alonzo.txwitsVKey' wits)
+    ++ Set.foldr ((:) . ShelleyBootstrapWitness era) [] (Ledger.Alonzo.txwitsBoot' wits)
  where
-  -- REVIEW(SN): The ledger's 'Addr' type is bigger than we need here and we
-  -- are forced to come up with a prefix for Byron "bootstrap" addresses,
-  -- although they should actually be serialised differently..and would not be
-  -- relevant for Hydra in the first place.
-  prefix = case addr of
-    (Cardano.Addr Cardano.Mainnet _ _) -> [Bech32.humanReadablePart|addr|]
-    (Cardano.Addr Cardano.Testnet _ _) -> [Bech32.humanReadablePart|addr_test|]
-    (Cardano.AddrBootstrap _) -> [Bech32.humanReadablePart|addr_boot|]
+  era =
+    ShelleyBasedEraAlonzo
 
 --
--- Output
+-- Formatting
```

Comment:
`prettyUtxo` would fit in here as well if you would like to sort them like this.